chain-extensions/token/src/lib.rs

```diff
@@ -219,7 +218,7 @@ where
 		"Calling transfer() sending {:?} {:?}, from {:?} to {:?}",
 		amount,
 		currency_id,
-		env.ext().caller().clone(),
+		&(*env.ext().caller()).clone(),
```

Comment:
About all these 'clone-double-ref' changes, I looked into what exactly this means and I guess the easiest change would be to just use `env.ext().caller()` each time, without any cloning, since the `.caller()` function already returns a reference. @gianfra-t Can you please try if changing all the `&(*env.ext().caller()).clone()` to just `env.ext().caller()` works? Maybe I'm missing something.

Reply:
Wow yes I missed that. I'm surprised clippy even suggested this vs just removing clone. But you are right it returns a reference to account id already, and all methods require a reference only.